### DISCUSSION TITLE
Possible to skip nodejs?

### DISCUSSION THREAD

**Discussion by Muvkina** [socketio/socket.io#4260] [Q&A] [answered]
Hello! I was looking at socketio and I am very interested for using it for some projects. However I am trying to figure out if it is possible to run the server in the browser without nodejs or any other server running on a local machine? The use case is I want users to be able to host a server in their browser without having to install anything since I cannot not afford servers to host and the project I am working on would work fine with people hosting their own servers(even if they have to port forward). My other question is if I can host a server purely in the browser let's say I didn't have an internet connection can someone else on the same network connect to me offline.

**Comment by darrachequesne** [comment, marked as answer]
Hi!

Socket.IO, like plain WebSockets, is designed for server to client communication.

That being said, you might be interested in WebContainers, which provide a way to run Node.js in the browser: https://blog.stackblitz.com/posts/introducing-webcontainers/

You may also check peer-to-peer communication and WebRTC, which sounds really close to your use case.

Reference: https://developer.mozilla.org/en-US/docs/Web/Guide/API/WebRTC/Peer-to-peer_communications_with_WebRTC

**Reply by Muvkina** [reply]
Thanks for the reply webrtc does seem to have the possibility to work for my use case. Do you know if it's still possible to "host" a server in nodejs? I like the idea of having both options one where users can just use their clients or connect to a server being hosted on node. Also is webrtc suitable for multi-player games? No worries if u don't feel like answering these due to not being related to socketio thanks for the response!

**Reply by darrachequesne** [reply]
> Do you know if it's still possible to "host" a server in nodejs?

I'm not sure to totally understand your use case, but I don't think you can run a Node.js server in your browser.

> Also is webrtc suitable for multi-player games?

Honestly, I don't know.

Good luck with your project!

**Reply by Muvkina** [reply]
No problem thanks for the help! I've done some digging it seems webrtc should be more then suitable for my needs and allows for the server to be run in browser and on node.
